libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1447,6 +1465,56 @@ string YulUtilFunctions::readFromStorageDynamic(Type const& _type, bool _splitFu
 		.render();
 	});
 }
+string YulUtilFunctions::readFromStorageReferenceType(Type const& _type, size_t _offset)
+{
+	solAssert(_type.category() == Type::Category::Struct, "");
+
+	string functionName =
+		"read_from_storage_offset_" +
+		to_string(_offset) +
+		"_" +
+		_type.identifier();
+
+	solAssert(_offset == 0, "");
```

Comment:
This should all move inside the lazy generator.

Reply:
I think this function should not have `_offset` parameter at all (reference types have implicit offset 0), removed it now.